Terraform aws ecs boomi

ahaffar opened the pull request:
- Restructure to terraform standard structure
 - remove stacks, providers folders
 - create main.tf, variables.tf and outputs.tf in the root folder

ahaffar committed:
updating gitignore

ahaffar committed:
Refactor to Terraform Standard Structure
 - Remove providers, modules and stacks dirs
 - Create a sub-module dir.
 - Create the main modules with variables and outputs.

ahaffar committed:
Updating README with an example

AdamiHamza commented on main.tf:
```diff
@@ -0,0 +1,92 @@
+  required_version = ">= 0.13.0"
+}
+
+provider "aws" {
```
Providers should not be created inside a reusable module and its creation is delegated to the caller module

AdamiHamza commented on main.tf:
```diff
@@ -0,0 +1,92 @@
+# Locals and Data Sources
+# ----------------------------------------------------------------------------------------------------------------------
+locals {
+  prefix = "${var.env}-${var.project_name}-${replace(data.aws_region.current.name, "-", "")}"
```
I think we should not prefix module resources with `var.env`, `var.project_name` and region because it's a reusable module.
Instead of that, we can delete `var.env`, `var.project_name` inputs and replace it with a `var.prefix` that can be specified by the caller module

AdamiHamza commented on main.tf:
```diff
@@ -0,0 +1,92 @@
+# KMS: KMS KEY ID used to encrypt the secrets
+# ---------------------------------------------------------------------------------------------------------------------
+
+module "ecs-cluster" {
```
We should add fmt

AdamiHamza commented on modules/ecs-cluster/main.tf:
```diff
@@ -0,0 +1,262 @@
+# ----------------------------------------------------------------------------------------------------------------------
+# S3 Logging Bucket
+# ----------------------------------------------------------------------------------------------------------------------
+resource "aws_s3_bucket" "bucket" {
```
Logging s3 bucket should be provided as an input to the reusable module, maybe the caller wants to use another s3 bucket instead of creating a new logging bucket just for Boomi.

ahaffar committed:
Fixing/Improivng the modules stractures
 - Removing local.prefix and add it to variables, so the user has the ability to add any needed string as prefix
 - Removing var.env - var.project which were used in local.prefix
 - Adding the s3_logging map(string) to required variables and removing the aws_s3_bucket resource
 - Creating a new folder /alb/ on the provided bucket_name which will be used later in ALB
 - Updating README.md with terraform-docs output for the new changes
 - Adding .tfvars to gitignore files
 - Fixing S3 folder object

kodless reviewed (approved):
LGTM

ahaffar committed:
Updating README.md usage example with the required variables

AdamiHamza reviewed (approved):
LGTM

ahaffar committed:
Fixing README
 - Fix README.md to remove extra variable
 - Change the ref tag version
